source/slang/slang.cpp

```diff
@@ -2629,6 +2635,33 @@ void EndToEndCompileRequest::init()
     m_frontEndReq = new FrontEndCompileRequest(getLinkage(), m_writers, getSink());
 }
 
+SlangResult EndToEndCompileRequest::_writeHash(ComponentType* component)
+{
+    DigestBuilder<SHA1> builder;
+
+    getLinkage()->buildHash(builder, 0);
+
+    // Enumerate all file dependencies and add them to the hash.
```

Comment:
Ought we include:

- Compiler options
- Compiler version
- Any environment variables